File: C/fibonacci.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int fibonacci (int N){  //Finds the nth fibonacci number
long int a=1;
long int b=1;
int c=1;
long int n;
if (N<1) {  //If the input number is not a positive integer
  printf("The Fibonacci %i doesn't exist\n", N);
}
else {
if (c==N){ //If N=1
printf("\nThe first Fibonacci number is 1\n");
}
else {
while(c<N){
n=a+b;
a=b;
b=n;
c++;
}
printf("\nThe %ith fibonacci number is: %li\n", N, a);
}
return 0;
}}

int nFibonacci(){
int N;
printf("What Fibonacci N do you want to know?: ");
scanf("%i", &N);
fibonacci (N);
return 0;
}

int main(){
nFibonacci();
return 0;
}

File: Java/fibonacci.java
import javax.swing.JOptionPane;

  public class fibonacci {
    
  public static int fibonacci (int N){
    int F=0;
     if (N==1){
       F=0;
    }
     if (N==2){
       F=1;
    }
     if (N>2) {
       F=fibonacci(N-1)+fibonacci(N-2);
    }
  return F;  
  } 
    
   public static void main(String[] args) {
     String num= JOptionPane.showInputDialog("What Fibonacci number you want to know?");
      int number= Integer.parseInt(num);
      int F=fibonacci(number);
    JOptionPane.showMessageDialog(null , "The " + number +" Fibonacci number is  "  + F );
   }}
